Bump version to 1.52.0 (#2421)
smithy-lang/smithy: CHANGELOG.md

# Smithy Changelog

## 1.52.0 (2024-10-15)

> ### Features

Reviewer: All the PR links aren't actual markdown links, they're only rendered that way on GitHub because it has a custom check for things that match issue/PR IDs.

Author: Forgot to do that, fixed.

* Added validator for identifiers missing references (#2418)
* Added checksum algorithm to model enum  (#2419)

> * Updated the httpChecksum trait spec (#2413)

Reviewer: ```suggestion
* Updated the `httpChecksum` trait spec (#2413)
```

* Added `breakingChanges` to the `httpChecksum` trait (#2398)
* Use the most common suffix for IDL inline IO (#2397)

### Bug Fixes

* Fixed trait codegen to properly support lists of enums (#2420)

> * Fixed API Gateway compatibility by requiring an Accept header for RPCv2 requests (#2417)

Reviewer: ```suggestion
* Update RPC v2 CBOR spec to require an Accept header for requests (#2417)
```

* Changed prefix header tests to allow for empty headers (#2415)

> * Fixed open-API plugin to properly support effective documentation precedence (#2402)

Reviewer: ```suggestion
* Fixed OpenAPI plugin to properly support effective documentation precedence (#2402)
```
